src/rwa/components/inputs/number-input.tsx

```diff
@@ -0,0 +1,60 @@
+import { ComponentPropsWithRef } from 'react';
+import { Control, Controller, FieldValues, Path } from 'react-hook-form';
+import { NumericFormat } from 'react-number-format';
+import { RWATableTextInput } from '.';
+
+type Props<ControlInputs extends FieldValues> = ComponentPropsWithRef<
+    typeof RWATableTextInput
+> & {
+    name: Path<ControlInputs>;
+    control: Control<ControlInputs>;
+    disabled?: boolean;
+    requiredErrorMessage?: string;
+    currency?: 'USD';
```

Comment:
for now 'USD' is the only one supported, but we can expand this list when needed